main.py

```diff
@@ -1,59 +1,125 @@
 import numpy as np
```

Comment:
just in general we need more helpful comments. Add 5 more block comments and 10 more inline comments. Make sure they are actually useful